opm/simulators/linalg/bda/WellContributions.cpp

```diff
@@ -57,7 +56,17 @@ WellContributions::~WellContributions()
 
     if(num_std_wells > 0){
         delete[] val_pointers;
+#if HAVE_OPENCL
+        delete[] h_toOrder;
+#endif
     }
+
+#if HAVE_OPENCL
+    if(num_ms_wells > 0){
+        delete[] h_x;
+        delete[] h_y;
+    }
+#endif
```

Comment:
I don't think this works. You might need to check the gpu mode. If both HAVE_OPENCL and HAVE_CUDA is 1 and we are running with `--gpu-mode=cusparse` these arrays might be freed twice as they are malloced with `CudaMallocHost` and freed with `CudaFreeHost` in `WellContributions.cu`.